src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -414,6 +418,82 @@ public Void clearEntitySet( UUID entitySetId ) {
         return null;
     }
 
+    @Override
+    @Timed
+    @RequestMapping(
+            path = { "/" + ENTITY_SET + "/" + SET_ID_PATH + "/" + ENTITY_KEY_ID_PATH + "/" + NEIGHBORS },
+            method = RequestMethod.DELETE
+    )
+    public Long clearEntityAndNeighborEntities(
+            @PathVariable( ENTITY_SET_ID ) UUID vertexEntitySetId,
+            @PathVariable( ENTITY_KEY_ID ) UUID vertexEntityKeyId
+    ) {
+
+        /*
+         * 1 - collect all relevant EntitySets and PropertyTypes
+         */
+
+        // getNeighborEntitySetIds() returns source, destination, and edge EntitySet ids
+        Set<UUID> neighborEntitySetIds = dgm.getNeighborEntitySetIds( vertexEntitySetId );
```

Comment:
Is this full copy necessary?

Reply:
allEntitySetIds = neighborEntitySetIds + vertexEntitySetId but is constructed by rehashing all neighborEntitySetIds, which returns a mutable set safe to use set. If you are concerned about safe guarding accidental mutation of the set ```Collections.unmodifiableSet(...)``` seems more appropriate in this case.